subtests/docker_cli/dockerinspect/test_inspect_keys.py

```diff
@@ -0,0 +1,270 @@
+import json
+
+
+class TestFilterKeys(TestCase):
+    def _run_one_test(self, spec):
+        """
+        Helper for running an individual test.
+        spec is a multi-line string containing specially-formed JSON.
+        The parsing rule is: by default, the expected result is the
+        same as the input. But if the '|' (pipe) character appears
+        on the line, we treat it that line as having a left- and
+        right-hand side. The left is our input JSON, the right is
+        the expected JSON after filtering. Most of the time the
+        right-hand side will be blank, indicating removed keys,
+        but it may also be used for keys/values that change names.
+        """
+        mockself = Mock(spec=inspect_keys.inspect_keys)
+
+        lhs = ''              # left-hand side: our original json
+        rhs = ''              # right-hand side: what we expect from filter
+        for line in spec.split("\n"):
+            divider = line.find(' |')
+            if divider >= 0:
+                lhs += line[:divider - 1] + "\n"
+                rhs += line[divider + 2:] + "\n"
+            else:
+                lhs += line + "\n"
+                rhs += line + "\n"
+
+        actual = inspect_keys.inspect_keys.filter_keys(mockself,
+                                                       json.loads(lhs))
+        self.maxDiff = None
+        self.assertEqual(actual, json.loads(rhs))
+
+    def test_basic(self):
+        """
+        Very simple test; makes sure that .Config.Labels is removed
+        """
+        self._run_one_test("""
+[
+{
+    "A": "B",
+    "Config": {
+       "Labels": { "a": [ "this gets deleted" ] },   | "Labels": {},
+       "Other":  [ "this is preserved" ]
+    }
+}
+]
+""")
+
+    def test_multi_net(self):
+        """
+        Multiple .NetworkSettings.Networks.xxx are renamed
+        """
+        self._run_one_test("""
+[
+{
+    "NetworkSettings": {
+        "Networks": {
+            "bridge": {                  | "RenamedForTesting_bridge": {
+                "EndpointID": "--preserved--"
+            },
+            "nextnet": {                 | "RenamedForTesting_nextnet": {
+                "AnotherPreservedKey": "yep"
+            },
+            "yetanothernet": {           | "RenamedForTesting_yetanothernet": {
+                "never": "mind"
+            },
+            "CamelCase": {               | "RenamedForTesting_CamelCase": {
+                "never": "mind"
+            }
+        }
+    }
+}
+]
+""")
+
+    def test_full(self):
+        """
+        Huge monster results, taken from a real-world instance.
+        """
+        self._run_one_test("""
+[
+{
+    "Id": "ed3b9325177ccd70717d58885dbeb53255b52f200be1dd7e65221e5eb805113e",
+    "Created": "2016-03-30T17:51:47.675978044Z",
+    "Path": "tail",
+    "Args": [
+        "-n1",
+        "/proc/1/cgroup"
+    ],
+    "State": {
+        "Status": "exited",
+        "Running": false,
+        "Paused": false,
```

Comment:
like this, ``false`` is from java, python is more sophisticated, it needs an upper case `F` :grinning: